### PR TITLE
Enabling package generation

### PR DESCRIPTION
**Pull request by aaronpowell**
Making it so we now generate packages on builds to make it easier to test

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -60,8 +59,8 @@ jobs:
...
           registry-url: https://npm.pkg.github.com/
-          scope: "@manekinekko"
-      - run: echo "registry=https://npm.pkg.github.com/@manekinekko" >> .npmrc
+          scope: "@azure"
+      - run: echo "registry=https://npm.pkg.github.com/@azure" >> .npmrc
       - run: npm publish $(ls *.tgz)
         env:
-          NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
\ No newline at end of file
+          NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (manekinekko) [inline .github/workflows/node.js.yml:66]: We probably need to regenerate a new token. I'll check if I have permissions.

Comment (aaronpowell) [reply .github/workflows/node.js.yml:66]: Not needed, that token is provided by GitHub Actions automagically

Comment (manekinekko) [reply .github/workflows/node.js.yml:66]: 😃

**Review by manekinekko — APPROVED** [review]

**Comment by manekinekko** [comment]
That's what I was thinking, it seems we can't publish the package that easily !
image

**Comment by aaronpowell** [comment]
I think the reason that the job failed is because _I_ don't have contribute access to the project so it's come from an external source, so the `GITHUB_TOKEN` isn't set to one that has the right scopes for the Azure org.

**Comment by manekinekko** [comment]
Pinging @anthonychu as I think he is working on a similar feature.

**Comment by anthonychu** [comment]
I’m working on publishing to npm. If this is publishing ci packages to GitHub registry I think that’s different.

**Comment by aaronpowell** [comment]
@anthonychu this is publishing to GitHub Packages, ala https://www.aaron-powell.com/posts/2020-11-06-deploy-to-github-packages-with-github-actions/

TL;DR - on each push it'll publish a pre-release version to GitHub Packages. But I'm not sure what impact that'd have with ESRP

**Comment by manekinekko** [comment]
@anthonychu do we want to publish to Github Packages as well? I am ok with either.

**Comment by manekinekko** [comment]
@aaronpowell @anthonychu what's the status of this PR?

**Comment by aaronpowell** [comment]
I'll defer to @anthonychu, but my gut feeling is that the signing process may pose a problem, as we'd have to get the package signed before it could go to GitHub Packages (and maybe even as an artifact of the build) since someone could download and use it.

**Comment by manekinekko** [comment]
I am closing this PR of now. We will open it if needed.
